kafka-ui-react-app/src/tests/Topics/Details/Messages/Messages.spec.tsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import JSONTree from 'react-json-tree';
+import * as useDebounce from 'use-debounce';
+import DatePicker from 'react-datepicker';
+import Messages, {
+  Props,
+} from '../../../../components/Topics/Details/Messages/Messages';
+import PageLoader from '../../../../components/common/PageLoader/PageLoader';
+
+describe('Messages component', () => {
+  beforeEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  const setupWrapper = (props: Partial<Props> = {}) => (
+    <Messages
+      clusterName="Test cluster"
+      topicName="Cluster topic"
+      isFetched
+      fetchTopicMessages={jest.fn()}
+      messages={[]}
+      partitions={[]}
+      {...props}
+    />
+  );
+
+  describe('Initial state', () => {
+    it('renders PageLoader', () => {
+      expect(
+        shallow(setupWrapper({ isFetched: false })).find(PageLoader)
+      ).toBeTruthy();
+    });
+  });
+
+  describe('Messages table', () => {
+    describe('With messages', () => {
+      const messagesWrapper = shallow(
+        setupWrapper({
+          messages: [
+            {
+              partition: 1,
+              offset: 2,
+              timestamp: new Date('05-05-1994'),
+              content: [1, 2, 3],
+            },
+          ],
+        })
+      );
+      it('renders table', () => {
+        expect(messagesWrapper.find('TimeStamp')).toBeTruthy();
+      });
+      it('renders JSONTree', () => {
+        expect(messagesWrapper.find(JSONTree)).toBeTruthy();
+      });
+      it('parses message content correctly', () => {
+        const messages = [
+          {
+            partition: 1,
+            offset: 2,
+            timestamp: new Date('05-05-1994'),
+            content: [1, 2, 3],
+          },
+        ];
+        const content = JSON.stringify(messages[0].content);
+        expect(JSON.parse(content)).toEqual(messages[0].content);
+      });
+    });
+    describe('Without messages', () => {
+      it('renders string', () => {
+        const wrapper = shallow(setupWrapper());
+        expect(wrapper.text()).toContain('No messages at selected topic');
+      });
+    });
+  });
+
+  describe('Offset field', () => {
+    describe('Seek Type dependency', () => {
+      const wrapper = mount(setupWrapper());
+
+      it('renders DatePicker', () => {
+        wrapper
+          .find('[id="selectSeekType"]')
+          .simulate('change', { target: { value: 'TIMESTAMP' } });
+
+        expect(
+          wrapper.find('[id="selectSeekType"]').first().props().value
+        ).toEqual('TIMESTAMP');
+
+        expect(wrapper.find(DatePicker)).toBeTruthy();
```

Comment:
looks wrong... Can we use something like 
```suggestion
        expect(wrapper.exists(DatePicker)).toBeTruthy();
```
or 

```suggestion
        expect(wrapper.find(DatePicker).length).toEqual(1);
```

Reply:
Please update all other examples